consensus_encoding/src/encode/mod.rs

```diff
@@ -91,6 +112,24 @@ impl<'s, T: Encodable + 's> Iterator for EncodableByteIter<'s, T> {
     }
 }
 
+impl<'s, T> ExactSizeIterator for EncodableByteIter<'s, T>
+where
+    T: Encodable + 's,
+    T::Encoder<'s>: ExactSizeEncoder
+{
+    fn len(&self) -> usize { self.enc.len() - self.position }
+}
+
+
+/// An encoder with a known size.
+pub trait ExactSizeEncoder: Encoder {
+    /// The number of bytes that the encoder yields.
```

Comment:
```suggestion
    /// The number of bytes remaining that the encoder will yield.
```

Or something like that.